README.md

```diff
@@ -1,27 +1,180 @@
+```
+
+If you need to use a specific version, replace [latest]{.title-ref} with
+the desired version tag:
+
+``` shell
+docker pull scylladb/cassandra-stress:3.17.0
+docker run --rm scylladb/cassandra-stress:3.17.0 <commands>
+```
+
+### Download Prebuilt Binaries
+
+You can download the prebuilt binaries directly from our GitHub
+releases.
+
+**Prerequisite**: Ensure that Java Runtime Environment (JRE) version 11
+or newer is installed on your system.
+
+**Using \`curl\`:**
+
+``` shell
+curl -LO https://github.com/scylladb/cassandra-stress/releases/download/v3.17.0/cassandra-stress-3.17.0-bin.tar.gz
+curl -LO https://github.com/scylladb/cassandra-stress/releases/download/v3.17.0/cassandra-stress-3.17.0-bin.tar.gz.sha256
+```
+
+**Using \`wget\`:**
+
+``` shell
+wget https://github.com/scylladb/cassandra-stress/releases/download/v3.17.0/cassandra-stress-3.17.0-bin.tar.gz
+wget https://github.com/scylladb/cassandra-stress/releases/download/v3.17.0/cassandra-stress-3.17.0-bin.tar.gz.sha256
+```
+
+**Verify the download using \`sha256sum\`:**
+
+``` shell
+sha256sum -c cassandra-stress-3.17.0-bin.tar.gz.sha256
+```
+
+If the verification is successful, you will see:
+
+``` text
+cassandra-stress-3.17.0-bin.tar.gz: OK
+```
+
+### Build from Source
+
+To build [cassandra-stress]{.title-ref} from source, ensure you have the
+following prerequisites installed: - **Java JDK**: Version 11 or newer -
+**Apache Ant**: Build tool
+
+Follow these steps:
+
+1. Clone the repository:
+
+    ``` shell
+    git clone --depth=1 https://github.com/scylladb/cassandra-stress.git
+    ```
+
+2. Checkout the specific version (3.17.0):
+
+    ``` shell
+    cd cassandra-stress
+    git checkout tags/3.17.0
+    ```
+
+3. Build the binaries using Apache Ant:
+
+    ``` shell
+    ant -Drelease=true artifacts
+    ```
+
+The compiled binaries will be located in the [build]{.title-ref}
+directory.
+
+### Build Docker Image from Source
+
+If you prefer a Dockerized build, follow these steps:
+
+1. Clone the repository:
+
+    ``` shell
+    git clone --depth=1 https://github.com/scylladb/cassandra-stress.git
+    ```
+
+2. Build the Docker image:
+
+    ``` shell
+    cd cassandra-stress
+    docker build -t cassandra-stress --compress .
+    ```
+
+Once built, you can run the image locally using:
+
+``` shell
+docker run --rm cassandra-stress <commands>
+```
 
 ## Usage
 
+### Examples
+
+``` console
+tools/bin/cassandra-stress write n=1000000 -node 192.168.1.101 # 1M inserts to given host
+tools/bin/cassandra-stress read n=10000000 -node 192.168.1.101 -o read # 1M reads
+tools/bin/cassandra-stress write -node 192.168.1.101,192.168.1.102 n=10000000 # 10M inserts spread across two nodes
+tools/bin/cassandra-stress help -pop # Print help for population distribution option
+```
+
+Or run it in a container, please don't miss the quotes around the command line:
+
+``` console
+docker run scylladb/cassandra-stress 'cassandra-stress write n=100000 -node 1.2.3.4'
```

Comment:
usage of  `--network=host --security-opt seccomp=unconfined`, should be mentioned 
otherwise you get quite high performance penalty, by using docker

Reply:
I'll add another block in the documentation with this option, and not as the default running parameter.